iis-wf/iis-wf-export-actionmanager/src/main/java/eu/dnetlib/iis/wf/export/actionmanager/relation/citation/CitationRelationExporterUtils.java

```diff
@@ -0,0 +1,170 @@
+        }
+
+        public String getDestinationDocumentId() {
+            return destinationDocumentId;
+        }
+
+        public void setDestinationDocumentId(String destinationDocumentId) {
+            this.destinationDocumentId = destinationDocumentId;
+        }
+
+        public Float getConfidenceLevel() {
+            return confidenceLevel;
+        }
+
+        public void setConfidenceLevel(Float confidenceLevel) {
+            this.confidenceLevel = confidenceLevel;
+        }
+    }
+
+    public static Dataset<Text> relationsToSerializedActions(Dataset<Relation> relations) {
+        return relations
+                .map(toSerializedActionMapFn(), Encoders.kryo(Text.class));
+    }
+
+    private static MapFunction<Relation, Text> toSerializedActionMapFn() {
+        return (MapFunction<Relation, Text>) payload ->
+                new Text(AtomicActionSerializationUtils.serializeAction(buildAtomicAction(payload), OBJECT_MAPPER));
+    }
+
+    private static AtomicAction<Relation> buildAtomicAction(Relation payload) {
+        return new AtomicAction<>(Relation.class, payload);
+    }
+
+    public static Dataset<ReportEntry> relationsToReportEntries(SparkSession spark, Dataset<Relation> relations) {
+        long totalRelationCount = totalRelationCount(relations);
+        long uniqueCitesRelationCount = uniqueCitesRelationCount(relations);
+        long uniqueIsCitedByRelationCount = uniqueIsCitedByRelationCount(relations);
+
+        return spark.createDataset(Arrays.asList(
+                ReportEntryFactory.createCounterReportEntry(REPORT_ENTRY_KEY_REFERENCES, totalRelationCount),
+                ReportEntryFactory.createCounterReportEntry(REPORT_ENTRY_KEY_CITES_DOCS, uniqueCitesRelationCount),
+                ReportEntryFactory.createCounterReportEntry(REPORT_ENTRY_KEY_ISCITEDBY_DOCS, uniqueIsCitedByRelationCount)
+        ), Encoders.kryo(ReportEntry.class));
+    }
+
+    private static long totalRelationCount(Dataset<Relation> relations) {
+        return relations
+                .filter((FilterFunction<Relation>) relation -> relation.getRelClass().equals(OafConstants.REL_CLASS_CITES))
+                .count();
+    }
+
+    private static long uniqueCitesRelationCount(Dataset<Relation> relations) {
+        return relations
+                .filter((FilterFunction<Relation>) relation -> relation.getRelClass().equals(OafConstants.REL_CLASS_CITES))
+                .map((MapFunction<Relation, String>) Relation::getSource, Encoders.STRING())
+                .distinct()
+                .count();
+    }
+
+    private static long uniqueIsCitedByRelationCount(Dataset<Relation> relations) {
+        return relations
+                .filter((FilterFunction<Relation>) relation -> relation.getRelClass().equals(OafConstants.REL_CLASS_ISCITEDBY))
+                .map((MapFunction<Relation, String>) Relation::getSource, Encoders.STRING())
+                .distinct()
+                .count();
+    }
```

Comment:
Minor optimization: we could simply reduce the number of lines almost by half after putting the common node within the method parameterized with `String relClass` and reuse it two places. This way we would also make sure the logic is aligned for both cases.